### PR TITLE
Retrieving and using a safe token id (with NaN value) for masking

### LINKED ISSUE #13: What happens if first vocab index is a number token?
This line induces undesired behavior then: https://github.com/ai4sd/number-token-loss/blob/55ba072e70687b73d40ef3772acf18f477177c3e/ntloss/core.py#L184

### PR DESCRIPTION
**Pull request by casinca**
Potential fix #13

### CODE REVIEW

**Review by jannisborn — CHANGES_REQUESTED** [review]
Thanks, great job, I would just raise explicitly, see my suggestion

**Review by jannisborn — APPROVED** [review]

File: ntloss/core.py [hunk 1]
```diff
@@ -70,6 +70,7 @@ def setup_number_tokens(self):
                     self.number_values[id] = float(token)
 
         self.is_number_token = ~torch.isnan(self.number_values)
+        self.nan_id = torch.where(~self.is_number_token)[0][0].item()
```
[the remark this answers is not in the document]
Comment (jannisborn) [reply ntloss/core.py:71]: Indeed this is faster but since it's in the constructor, I would pick code cleanliness and readability over speed
